lib/shoulda/matchers/active_record/have_encrypt_attribute_matcher.rb

```diff
@@ -0,0 +1,160 @@
+    module Matchers
+      module ActiveRecord
+        # The `have_encrypt_attribute` matcher tests usage of the
+        # `encrypts` macro (Rails 7+ only).
+        #
+        #     class Survey < ActiveRecord::Base
+        #       encrypts :access_code
+        #     end
+        #
+        #     # RSpec
+        #     RSpec.describe Survey, type: :model do
+        #       it { should have_encrypt_attribute(:access_code) }
+        #     end
+        #
+        #     # Minitest (Shoulda)
+        #     class SurveyTest < ActiveSupport::TestCase
+        #       should have_encrypt_attribute(:access_code)
+        #     end
+        #
+        # #### Qualifiers
+        #
+        # ##### deterministic
+        #
+        #     class Survey < ActiveRecord::Base
+        #       encrypts :access_code, deterministic: true
+        #     end
+        #
+        #     # RSpec
+        #     RSpec.describe Survey, type: :model do
+        #       it { should have_encrypt_attribute(:access_code).deterministic(true) }
+        #     end
+        #
+        #     # Minitest (Shoulda)
+        #     class SurveyTest < ActiveSupport::TestCase
+        #       should have_encrypt_attribute(:access_code).deterministic(true)
+        #     end
+        #
+        # ##### downcase
+        #
+        #     class Survey < ActiveRecord::Base
+        #       encrypts :access_code, downcase: true
+        #     end
+        #
+        #     # RSpec
+        #     RSpec.describe Survey, type: :model do
+        #       it { should have_encrypt_attribute(:access_code).downcase(true) }
+        #     end
+        #
+        #     # Minitest (Shoulda)
+        #     class SurveyTest < ActiveSupport::TestCase
+        #       should have_encrypt_attribute(:access_code).downcase(true)
+        #     end
+        #
+        # ##### ignore_case
+        #
+        #     class Survey < ActiveRecord::Base
+        #       encrypts :access_code, deterministic: true, ignore_case: true
+        #     end
+        #
+        #     # RSpec
+        #     RSpec.describe Survey, type: :model do
+        #       it { should have_encrypt_attribute(:access_code).ignore_case(true) }
+        #     end
+        #
+        #     # Minitest (Shoulda)
+        #     class SurveyTest < ActiveSupport::TestCase
+        #       should have_encrypt_attribute(:access_code).ignore_case(true)
+        #     end
+        #
+        # @return [HaveEncryptAttributeMatcher]
+        #
+        def have_encrypt_attribute(value)
+            HaveEncryptAttributeMatcher.new(value)
+        end
+  
+        # @private
+        class HaveEncryptAttributeMatcher
+          def initialize(attribute)
+            @attribute = attribute.to_sym
+            @options = {}
+          end
+  
+          attr_reader :failure_message, :failure_message_when_negated
+          
+          def deterministic(deterministic)
+            @options[:deterministic] = deterministic
+            self
+          end

```

Comment:
What do you think about extracting that to a method? The same pattern is used across the options.

Something like this:
```
def with_option(option_name, value)
  @options[option_name] = value
  self
end

def deterministic(deterministic)
  with_option(:deterministic, deterministic)
end
```